src/lib/pmd/PmdCatalogWrapper.ts

```diff
@@ -151,32 +149,20 @@ export class PmdCatalogWrapper extends PmdSupport {
 		return path.join(PMD_LIB, "pmd-" + language + "-" + PMD_VERSION + ".jar");
 	}
 
-	/**
-	 * Accepts a child process created by child_process.spawn(), and a Promise's resolve and reject function.
-	 * Resolves/rejects the Promise once the child process finishes.
-	 * @param cp
-	 * @param res
-	 * @param rej
-	 */
-	protected monitorChildProcess(cp: ChildProcessWithoutNullStreams, res: (string) => void, rej: (string) => void): void {
-		let stdout = '';
-
-		// When data is passed back up to us, pop it onto the appropriate string.
-		cp.stdout.on('data', data => {
-			stdout += data;
-		});
-
-		// When the child process exits, if it exited with a zero code we can resolve, otherwise we'll reject.
-		cp.on('exit', code => {
-			this.outputProcessor.processOutput(stdout);
-			this.logger.trace(`monitorChildProcess has received exit code ${code}`);
-			if (code === 0) {
-				res(stdout);
-			} else {
-				rej(messages.getMessage('error.external.errorMessageAbove'));
-			}
-		});
+	protected isSuccessfulExitCode(code: number): boolean {
+		// By internal convention, 0 indicates success and any non-0 code indicates failure.
+		return code === 0;
 	}
 
+	protected handleResults(args: ResultHandlerArgs): void {
```

Comment:
Implementation of new abstract method, retains unique behavior from old version of `monitorChildProcess`